internal/app/querymapping/provider.go

```diff
@@ -0,0 +1,105 @@
+package querymapping
+
+import (
+	"encoding/json"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"path"
+
+	"github.com/rs/zerolog/log"
+)
+
+type (
+	Provider struct {
+		queryMappings    []QueryMap
+		queryMappingPath string
+		tmpDir           string
+	}
+)
+
+const (
+	tmpDirPrefix = "query_mapping_"
+	fileName     = "mapping.json"
+)
+
+func NewProvider(queryMappingPath string) (*Provider, error) {
```

Comment:
Please add unit tests for this provider.

Reply:
Ok